### PR TITLE
4 - Add Z shaped block

### PR DESCRIPTION
**Pull request by bearyogi**
Changes:

- Added Z - shaped block with magenta color.
- Added JavaDoc for ZBlock class.
- Added tests for Z - shaped block class.
- Added Z - shaped block to BlockFeed.
- Added ZBlock test class to BlockFeedTestFactory

### CODE REVIEW

File: src/test/resources/testng.xml [hunk 1]
```diff
@@ -15,6 +15,8 @@
         </groups>
         <classes>
             <class name="com.epam.prejap.tetris.block.LBlockTest"/>
+            <class name="com.epam.prejap.tetris.block.IBlockTest"/>
```
Comment (pkierat-epam) [inline src/test/resources/testng.xml:18]: The PR is about "Z" block, and yet you add "I" block here? :raised_eyebrow:

Comment (bearyogi) [reply src/test/resources/testng.xml]: Corrected

**Comment by pkierat-epam** [comment]
Latest changes to be pulled, conflicts to be resolved, version to be bumped before we can proceed.

**Comment by bearyogi** [comment]
@pkierat-epam conflicts corrected.

**Comment by bearyogi** [comment]
@pkierat-epam Changes merged and adjusted, conflicts corrected.

**Comment by pkierat-epam** [comment]
Due to the value brought by Veronika's PR for tests, we decided to merge her version. That means that in the current state, your changes become obsolete. But fear not, you still can save it, by improving her solution as a part of "next iteration" (and I ensure you, there are still things to be improved there).

If you, however, decide not to improve it, please close this PR.

**Pull request closed without merging**